Python/Python.sublime-syntax

```diff
@@ -402,6 +402,120 @@ contexts:
       pop: true
 
   comments:
+    # Type-ignore can only be used by itself.
+    - match: '(#)\s*(type)(:)\s*(ignore)\b'
+      scope: comment.line.type-hint.python
+      captures:
+        1: punctuation.definition.comment.python comment.line.number-sign.python
+        2: keyword.other.annotation.type-comment.python
+        3: punctuation.separator.annotation.type-comment.python
+        4: keyword.other.annotation.type-comment.ignore.python
+      set: comments_afterignore
+    - match: '(#)\s*(type)(:)'
+      scope: comment.line.type-hint.python
+      captures:
+        1: punctuation.definition.comment.python comment.line.number-sign.python
+        2: keyword.other.annotation.type-comment.python
+        3: punctuation.separator.annotation.type-comment.python
+      embed: type-comment
+      escape: '\n'
+      embed_scope: comment.line.type-hint.python
+
+    - include: comments_noannotation
+
+  comments_afterignore:
+    - meta_scope: comment.line.type-hint.python
+    - include: comments_noannotation
+    - match: \n
+      pop: true
+    - match: \[
+      # Mypy's # type: ignore[error-code1, error-code2]
+      scope: punctuation.section.brackets.begin.python
+      set:
+        - meta_scope: comment.line.type-hint.python
+        - match: \[
+          scope: invalid.illegal.python
+        - match: '[a-zA-Z_-][a-zA-Z_0-9-]+'
+          scope: comment.line.type-hint.python.mypy-errorcode
```

Comment:
We don't need the `.mypy-errorcode` here. The order is reversed, but mostly it is far too specific a selector and it unlikely to really ever be used.